### PR TITLE
Hetzner provider

### PR DESCRIPTION
**Pull request by matthiasng**
PR for #15 15

### CODE REVIEW

File: hetzner/provider.go [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+import (
+	"context"
+	"strings"
+
+	"github.com/libdns/libdns"
+)
+
+// Provider implements the libdns interfaces for Hetzner
+type Provider struct {
+	//Client
```
Comment (mholt) [inline hetzner/provider.go:12]: Possible cleanup needed on aisle 12 :)

**Review by mholt — APPROVED** [review]
This actually looks pretty good from what I can see!

I'm creating the hetzner repo and adding you to it so you can push to it and maintain this. Thanks again!

Unrelated, but if you'd be interested in helping all CertMagic + Caddy users be able to use this, a simple wrapper over this provider could be created in the caddy-dns organization: https://github.com/caddy-dns -- would you be able to do that too? If so I will create that repo and add you to it as well. :) They're pretty simple; the "hardest" thing is just Caddyfile support which you can mostly copy-paste from any other repo. Anyway, you're totally not obligated to do this. Just that most users here tend to also be Caddy users right now so I thought I'd ask.

**Comment by matthiasng** [comment]
I'm always glad to help. Just create the caddy repo and send me an invite. I will do the rest ;)

**Comment by mholt** [comment]
Thanks! Done

**Pull request closed without merging**